doc/controller_configuration/controller_configuration_tutorial.rst

```diff
@@ -30,146 +79,407 @@ The ``controllers.yaml`` configuration file is located in the ``robot_moveit_con
-    allowed_goal_duration_margin: 0.5
+Additional options for tuning the behavior and safety checks of MoveIt's execution pipeline can be configured in ``trajectory_execution.launch.xml`` file generated by MSA:
 
-Create the Controller launch file
----------------------------------
-Now, create the controller launch file (call it ``robot_moveit_controller_manager.launch.xml`` where ``robot`` is the name of your robot as specified when you created your MoveIt robot config package).
+* ``execution_duration_monitoring``: when ``false``, will not throw error if a controller takes longer than expected to complete a trajectory.
+* ``allowed_goal_duration_margin``: same as above, but configured globally as a default for all controllers.
+* ``allowed_start_tolerance``: joint state tolerance when validating that a trajectory's first point matches current robot state. If set to ``0`` MoveIt will skip waiting for the robot to stop after execution.
 
-Add the following lines to this file: ::
+ROS Control Controller Manager
+------------------------------
 
- <launch>
-  <!-- Set the param that trajectory_execution_manager needs to find the controller plugin -->
-  <arg name="moveit_controller_manager" default="moveit_simple_controller_manager/MoveItSimpleControllerManager" />
-  <param name="moveit_controller_manager" value="$(arg moveit_controller_manager)"/>
-  <!-- load controller_list -->
-  <rosparam file="$(find robot_moveit_config)/config/controllers.yaml"/>
- </launch>
+Alternatively to the simple controller manager described above, MoveIt also provides a controller manager that directly interfaces the `ROS Controller Manager <http://wiki.ros.org/controller_manager>`_. Instead of using a bridging configuration file like ``simple_moveit_controllers.yaml``, this controller manager directly queries the ROS Controller Manager for available controllers.
 
-MAKE SURE to replace ``robot_moveit_config`` with the correct name of your MoveIt robot config package.
+The *MoveIt ROS Control Controller Manager* keeps track of all loaded and started ROS controllers, as well as the subset of these controllers that can be used with MoveIt. All loaded and started controllers are designated as *active*, and the subset of these controllers that can be used with MoveIt because they have associated controller handles is designated as *managed*.
 
-Now, you should be ready to have MoveIt talk to your robot.
+Multi Controller Manager
+------------------------
 
-Debugging Information
----------------------
-The ``FollowJointTrajectory`` or ``GripperCommand`` interfaces on your robot must be communicating in the namespace: ``/name/action_ns``. In the above example, you should be able to see the following topics (using *rostopic list*) on your robot:
+The *MoveIt ROS Control Controller Manager* can only interface controllers from the single ROS controller manager found in the ROS namespace defined by the ROS parameter ``~ros_control_namespace`` (defaults to ``/``).
```

Comment:
It was not clear to me where to put this parameter and it belongs to the previous section?

Reply:
I agree that this part assumes you are very comfortable with ROS configuration.

The "~" means root or top level (you could say "global") ROS configuration namespace. There is a number of ways to set global parameters, such as specifying them as command line arguments, putting them in launch files, loading them from YAML, or setting them via command line CLI commands.

This parameter is being read by `Controller_Manager_Plugin.cpp`:

```
/**
 * \brief The default constructor. Reads the namespace from ~ros_control_namespace param and defaults to /
 */
MoveItControllerManager()
  : ns_(ros::NodeHandle("~").param("ros_control_namespace", std::string("/")))
  , loader_("moveit_ros_control_interface", "moveit_ros_control_interface::ControllerHandleAllocator")
{
  ROS_INFO_STREAM("Started moveit_ros_control_interface::MoveItControllerManager for namespace " << ns_);
}
```

The easiest way to set it would probably be in `ros_controllers.launch` launch file by using `param` tag, but you could also pick any other launch file where you think it makes the most sense.

```
<param name="ros_control_namespace" value="myrobotcontrol" />
```